virtual_rainforest/core/grid.py

```diff
@@ -446,3 +452,132 @@ def populate_distances(
+        #    object https://shapely.readthedocs.io/en/latest/strtree.html
+
+        cell_map = [
+            [id for id, ply in zip(self.cell_id, self.polygons) if ply.intersects(pt)]
+            for pt in xyp
+        ]
+
+        return cell_map
+
+    def map_xy_to_cell_indexing(
+        self,
+        x_coords: np.ndarray,
+        y_coords: np.ndarray,
+        x_idx: Optional[np.ndarray],
+        y_idx: Optional[np.ndarray],
+        strict: Optional[bool] = True,
+    ) -> tuple[np.ndarray, np.ndarray]:
+        """Returns indexing to map xy coordinates a single cell_id axis.
+
+        This function maps the provided one-dimensional set of x and y points onto the
+        grid (using `~virtual_rainforest.core.grid.Grid.map_xy_to_cell_id`) and then
+        checks that the mapped points uniquely identify all of the grid cells. If
+        `strict` is set to False, then extra points outside the grid are permitted.
+
+        The function then returns a pair of arrays that give indices on the original
+        x and y data to extract data along a single cell id axis. If specific x and y
+        indexing data is not provided, they are assumed to be simple sequences along
+        `x_coords` and `y_coords`.
+
+        Args:
+            cell_map: An existing mapping from points onto cell ids.
+            strict: Should points outside the grid be allowed
+
+        Returns:
+            A list showing the integer cell id for each pair of points. If strict is
+            False, the list element for points outside the grid is None.
+        """
+
+        # Get the coordinate mapping to cell ids
+        cell_map = self.map_xy_to_cell_id(x_coords=x_coords, y_coords=y_coords)
+
+        # Set indexing to sequence along coords if missing
+        if (x_idx is None) and (y_idx is None):
+            _x_idx = np.arange(x_coords.shape[0])
+            _y_idx = np.arange(y_coords.shape[0])
+        elif (x_idx is None) ^ (y_idx is None):  # Note: ^ is xor
+            raise ValueError("Only one of x/y indices provided.")
+        else:
+            _x_idx = x_idx  # type: ignore
+            _y_idx = y_idx  # type: ignore
+
+        # Check provided
+        if (_x_idx.shape != x_coords.shape) or (  # type: ignore
+            _y_idx.shape != y_coords.shape  # type: ignore
+        ):  # type: ignore
+            raise ValueError("Dimensions of x/y indices do not match coordinates")
+
+        # Find the set of total number of cell mappings per point
+        cell_counts = set([len(mp) for mp in cell_map])
+
+        # If strict, raise an exception where not all coords fall in a grid cell
+        if strict and (0 in cell_counts):
+            raise ValueError("Mapped points fall outside grid.")
+
+        # Values greater than 1 indicate coordinates on cell edges
+        if any([c > 1 for c in cell_counts]):
+            raise ValueError("Mapped points fall on cell boundaries.")
```

Comment:
See comment above re cell boundary question

Reply:
Yes. If an XY point maps onto a boundary, then `cell_count > 1` and the data mapping fails.